File: haskell/euler/3.hs
sqrt2 x = floor(x ** 0.5)

isPrime :: Integer->Bool
isPrime num = null [x | x <- [2..sqrt2 (fromIntegral num)], mod num x == 0]

factors num = [x | x <- [2..floor( (fromIntegral num) *0.5)], mod num x == 0]
output = max (filter isPrime (factors 600851475143))

File: julia/euler/3.jl
square_rt = x -> x ^ 0.5
num = 600851475143
function is_prime_factor(y)
  (600851475143 % y == 0) && isprime(int(y))
end


possible_factors = [2:ceil(square_rt(num))]
println(filter(is_prime_factor, possible_factors)[end])
